### PR TITLE
Adding simple test for Dead letter sink and retries

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

## Proposed Changes

- Adding simple rekt-test for Dead letter sink and retries

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: test/e2e_new/dls_test.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	f.Setup("install source", eventshub.Install(
+		sourceName,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(ev),
+	))
+
+	f.Assert("receive event on sink",
+		OnStore(sinkName).
+			MatchEvent(cetest.HasId(ev.ID())).
+			Exact(1),
```
[the remark this answers is not in the document]
Comment (matzew) [reply test/e2e_new/dls_test.go:105]: yes

**Review by aavarghese — APPROVED** [review]
/lgtm

**Pull request merged**